plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1399 @@
+                        if matched_user:
+                            user_name = want['name']
+                            user_url = "data/ietf-snmp:snmp/usm/local/user={0}".format(user_name)
+                            auth = None
+                            priv = None
+                            priv_key = None
+                            auth_key = None
+                            if auth and auth_key and priv and priv_key and encrypted:
+                                user_request = {"path": user_url, "method": DELETE}
+                                user_requests.append(user_request)
+                            else:
+                                if want.get('auth') and 'auth_type' in want.get('auth'):
+                                    auth = want.get('auth').get('auth_type')
+                                if want.get('priv') and 'priv_type' in want.get('priv'):
+                                    priv = want.get('priv').get('priv_type')
+                                if want.get('auth') and 'key' in want.get('auth'):
+                                    auth_key = want.get('auth').get('key')
+                                if want.get('priv') and 'key' in want.get('priv'):
+                                    priv_key = want.get('priv').get('key')
+                                encrypted = want.get('encrypted')
+                                if auth and auth_key or auth:
+                                    if auth == 'sha2-256':
+                                        auth = 'ietf-snmp-ext:sha2-256'
+                                    elif auth == 'sha2-384':
+                                        auth = 'ietf-snmp-ext:sha2-384'
+                                    elif auth == 'sha2-512':
+                                        auth = 'ietf-snmp-ext:sha2-512'
+                                    auth_url = "{0}/auth/{1}".format(user_url, auth)
+                                    user_request = {"path": auth_url, "method": DELETE}
+                                    user_requests.append(user_request)
+                                if priv and priv_key or priv:
+                                    priv_url = "{0}/priv/{1}".format(user_url, priv)
+                                    user_request = {"path": priv_url, "method": DELETE}
+                                    user_requests.append(user_request)
+                                if encrypted:
+                                    encrypted_url = user_url + "/ietf-snmp-ext:encrypted"
+                                    user_request = {"path": encrypted_url, "method": DELETE}
+                                    user_requests.append(user_request)
+
+                            group_name = want.get('group')
+                            if group_name and have_group:
+                                group_url = "data/ietf-snmp:snmp/vacm/group={0}/memeber={0}".format(group_name)
```

Comment:
There is a typo in the URL specification ('memeber' instead of 'member') and the group member name is incorrect. (It is the group name instead of the name of this user.)

```suggestion
                                group_url = "data/ietf-snmp:snmp/vacm/group={0}/member={0}".format(user_name)
```